### ISSUE TITLE
JarFileResource#exists() does not work in spring-boot

### ISSUE THREAD

**Issue by tsachev** [eclipse/jetty.project#518]
When spring boot application is started from a fat jar/war keystore resources from
classpath may have following form

jar:file:/home/tsachev/nemesis/demos/conf-constr/target/demo-0.0.1-SNAPSHOT-boot.jar!/BOOT-INF/classes!/keystore.p12
or
jar:file:/home/tsachev/nemesis/demos/conf-constr/target/demo-0.0.1-SNAPSHOT-boot.war!/WEB-INF/classes!/keystore.p12

This worked with jetty 9.2 because only get input stream is called for keystore resources,
but in jetty 9.3 `exists()` is called first  and it returns false, even though
`getInputStream()` works correctly.

Looking at the source code of `JarFileResource` I found that if this line
https://github.com/eclipse/jetty.project/blob/master/jetty-util/src/main/java/org/eclipse/jetty/util/resource/JarFileResource.java#L124
is changed from
```java
int sep = _urlString.indexOf("!/");
```
to
```java
int sep = _urlString.lastIndexOf("!/");
```
it works as expected.

Do you think this is reasonable to change?

**Comment by gregw** [comment]
@tsachev if those are legal jar:file URLs, then I'm guessing it is a reasonable change.   But are they really legal?   Why the two !/ patterns?     The jar is `file:/home/tsachev/nemesis/demos/conf-constr/target/demo-0.0.1-SNAPSHOT-boot.war`   So the first !/ should indicate the resource within that jar, namely `WEB-INF/classes!/keystore.p12`. So what is the !/ in that path?

Taking the lastIndexOf would suggest that the jar file is actually `file:/home/tsachev/nemesis/demos/conf-constr/target/demo-0.0.1-SNAPSHOT-boot.war!/WEB-INF/classes`   ???  Is that true?  Does Spring-boot make such a strangely named zip file?

**Comment by tsachev** [comment]
@gregw I cannot really say if they are *legal* or not. But no they are not making such archives.

Spring boot executable jars have some structure that is basically
- a regular jar with their loader classes.
- the application classes are placed in the above archive under`BOOT-INF/classes` or `WEB-INF/classes`
- dependencies are in `BOOT-INF/lib` or `WEB-INF/lib` and `WEB-INF/lib-provided`.

The `war` variant should be compatible with the standard `war` structure while the `jar` variant is something similar to it. What spring boot does is that they want the application to be run from jar/war archive *without actually extracting it*.
More info on the structure and the reasoning behind it can be found in their reference documentation
http://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#executable-jar (relevant sections are D.1 and D.2)

The key implementation detail is that they override the default jar url protocol handler with their own
`JarURLConnection` that extends from `java.net.JarURLConnection` and also their own `JarFile` and `JarEntry` implementations that extends from the same classes in `java.util.jar` package.

**Comment by tsachev** [comment]
The `WEB-INF/classes` actually references to the nested jar with application code. It is however not really in actual nested jar. I suspect that this way they avoid confusion (an archive which contains another   archive with the same name) and also ensure that application classes will be the first in their class loader.

**Comment by joakime** [comment]
I suspect that Spring has added support for doubly-nested jar resources via their custom JarURLConnection

aka:

    jar:file:/path/to/a.jar!/path/to/b.jar!/path/to/resource.dat

Doubly-nested jars are not something that the Java JVM supports natively (rightfully so, as unpacking that is not great for performance)

**Comment by tsachev** [comment]
Yeah the performance of their loader is not great. For example there is noteable difference between startup time of a fat jar vs exploaded one. For production l would not use a fat jar, because of that, but it is one of the cool features of spring boot that makes the apps easy to start and update. you deal with just one executable file.

**Comment by gregw** [comment]
@joakime I thought that to, but the form of the URI posted by @tsachev is not like that, it is:
`jar:file:/home/tsachev/nemesis/demos/conf-constr/target/demo-0.0.1-SNAPSHOT-boot.jar!/BOOT-INF/classes!/keystore.p12`   So it is a classes directory within the jar?

So something else is going on???

Having said that, using `_urlString.lastIndexOf("!/");` is probably more technically correct anyway, so I'm happy to make the change.... just would be good to know what magic is happening under the springboot hood?

**Comment by tsachev** [comment]
@gregw @joakime You are both correct. The 'magic' that happens is in their extension of `java.util.jar.JarFile`.
https://github.com/spring-projects/spring-boot/blob/master/spring-boot-tools/spring-boot-loader/src/main/java/org/springframework/boot/loader/jar/JarFile.java#L252-L257

Basically spring-boot loader can create a "jar file" from both nested archive and a directory withing the archive.

Here is an output from `unzip -l` for original and repackaged wars for a sample spring boot app

```
Archive:  target/spring-boot-sample-jetty93-jsp-1.4.0.BUILD-SNAPSHOT.war.original
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  2016-04-25 09:25   META-INF/
      333  2016-04-25 09:25   META-INF/MANIFEST.MF
        0  2016-04-25 09:25   WEB-INF/
        0  2016-04-25 09:25   WEB-INF/lib/
        0  2016-04-25 09:25   WEB-INF/classes/
        0  2016-04-25 09:25   WEB-INF/classes/sample/
        0  2016-04-25 09:25   WEB-INF/classes/sample/jetty93/
        0  2016-04-25 09:25   WEB-INF/classes/sample/jetty93/jsp/
        0  2016-04-25 09:25   WEB-INF/jsp/
    64804  2016-04-18 10:03   WEB-INF/lib/classmate-1.3.1.jar
   804737  2016-04-25 09:20   WEB-INF/lib/spring-web-4.3.0.BUILD-SNAPSHOT.jar
    16431  2016-04-07 00:07   WEB-INF/lib/jcl-over-slf4j-1.7.21.jar
    50897  2016-04-07 00:08   WEB-INF/lib/jackson-annotations-2.7.3.jar
    63777  2015-06-02 14:57   WEB-INF/lib/validation-api-1.1.0.Final.jar
   601396  2016-04-25 09:21   WEB-INF/lib/spring-boot-1.4.0.BUILD-SNAPSHOT.jar
     2330  2016-04-25 09:22   WEB-INF/lib/spring-boot-starter-web-1.4.0.BUILD-SNAPSHOT.jar
    41071  2016-04-07 00:07   WEB-INF/lib/slf4j-api-1.7.21.jar
   414240  2014-02-08 15:50   WEB-INF/lib/jstl-1.2.jar
   304075  2016-04-07 00:07   WEB-INF/lib/logback-classic-1.1.7.jar
   263550  2016-04-25 09:20   WEB-INF/lib/spring-expression-4.3.0.BUILD-SNAPSHOT.jar
     4597  2016-04-07 00:08   WEB-INF/lib/jul-to-slf4j-1.7.21.jar
   273599  2016-02-26 23:10   WEB-INF/lib/snakeyaml-1.17.jar
   752725  2016-04-25 09:20   WEB-INF/lib/spring-beans-4.3.0.BUILD-SNAPSHOT.jar
  1110935  2016-04-25 09:20   WEB-INF/lib/spring-core-4.3.0.BUILD-SNAPSHOT.jar
   252518  2016-04-07 00:08   WEB-INF/lib/jackson-core-2.7.3.jar
     2295  2016-04-25 09:21   WEB-INF/lib/spring-boot-starter-logging-1.4.0.BUILD-SNAPSHOT.jar
  1128985  2016-04-25 09:20   WEB-INF/lib/spring-context-4.3.0.BUILD-SNAPSHOT.jar
   855505  2016-04-25 09:21   WEB-INF/lib/spring-boot-autoconfigure-1.4.0.BUILD-SNAPSHOT.jar
     2275  2016-04-25 09:21   WEB-INF/lib/spring-boot-starter-1.4.0.BUILD-SNAPSHOT.jar
     2287  2016-04-25 09:22   WEB-INF/lib/spring-boot-starter-validation-1.4.0.BUILD-SNAPSHOT.jar
    66802  2015-07-10 00:14   WEB-INF/lib/jboss-logging-3.3.0.Final.jar
   470782  2016-04-07 00:07   WEB-INF/lib/logback-core-1.1.7.jar
  1202276  2016-04-07 00:08   WEB-INF/lib/jackson-databind-2.7.3.jar
   704465  2016-02-24 21:13   WEB-INF/lib/hibernate-validator-5.2.4.Final.jar
    23646  2016-04-07 00:14   WEB-INF/lib/log4j-over-slf4j-1.7.21.jar
   905905  2016-04-25 09:20   WEB-INF/lib/spring-webmvc-4.3.0.BUILD-SNAPSHOT.jar
   379313  2016-04-25 09:20   WEB-INF/lib/spring-aop-4.3.0.BUILD-SNAPSHOT.jar
     1186  2016-04-25 09:25   WEB-INF/classes/sample/jetty93/jsp/WelcomeController.class
     1318  2016-04-25 09:25   WEB-INF/classes/sample/jetty93/jsp/SampleJetty93JspApplication.class
      168  2016-04-25 09:25   WEB-INF/classes/application.properties
      409  2016-04-25 09:19   WEB-INF/jsp/welcome.jsp
     2746  2016-04-25 09:19   META-INF/maven/org.springframework.boot/spring-boot-sample-jetty93-jsp/pom.xml
      155  2016-04-25 09:25   META-INF/maven/org.springframework.boot/spring-boot-sample-jetty93-jsp/pom.properties
---------                     -------
 10772533                     43 files
```

```
Archive:  target/spring-boot-sample-jetty93-jsp-1.4.0.BUILD-SNAPSHOT.war
warning [target/spring-boot-sample-jetty93-jsp-1.4.0.BUILD-SNAPSHOT.war]:  6965 extra bytes at beginning or within zipfile
  (attempting to process anyway)
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  2016-04-25 09:25   META-INF/
      564  2016-04-25 09:25   META-INF/MANIFEST.MF
        0  2016-04-25 09:25   WEB-INF/
        0  2016-04-25 09:25   WEB-INF/lib/
        0  2016-04-25 09:25   WEB-INF/classes/
        0  2016-04-25 09:25   WEB-INF/classes/sample/
        0  2016-04-25 09:25   WEB-INF/classes/sample/jetty93/
        0  2016-04-25 09:25   WEB-INF/classes/sample/jetty93/jsp/
        0  2016-04-25 09:25   WEB-INF/jsp/
    64804  2016-04-18 10:03   WEB-INF/lib/classmate-1.3.1.jar
   804737  2016-04-25 09:20   WEB-INF/lib/spring-web-4.3.0.BUILD-SNAPSHOT.jar
    16431  2016-04-07 00:07   WEB-INF/lib/jcl-over-slf4j-1.7.21.jar
    50897  2016-04-07 00:08   WEB-INF/lib/jackson-annotations-2.7.3.jar
    63777  2015-06-02 14:57   WEB-INF/lib/validation-api-1.1.0.Final.jar
   601396  2016-04-25 09:21   WEB-INF/lib/spring-boot-1.4.0.BUILD-SNAPSHOT.jar
     2330  2016-04-25 09:22   WEB-INF/lib/spring-boot-starter-web-1.4.0.BUILD-SNAPSHOT.jar
    41071  2016-04-07 00:07   WEB-INF/lib/slf4j-api-1.7.21.jar
   414240  2014-02-08 15:50   WEB-INF/lib/jstl-1.2.jar
   304075  2016-04-07 00:07   WEB-INF/lib/logback-classic-1.1.7.jar
   263550  2016-04-25 09:20   WEB-INF/lib/spring-expression-4.3.0.BUILD-SNAPSHOT.jar
     4597  2016-04-07 00:08   WEB-INF/lib/jul-to-slf4j-1.7.21.jar
   273599  2016-02-26 23:10   WEB-INF/lib/snakeyaml-1.17.jar
   752725  2016-04-25 09:20   WEB-INF/lib/spring-beans-4.3.0.BUILD-SNAPSHOT.jar
  1110935  2016-04-25 09:20   WEB-INF/lib/spring-core-4.3.0.BUILD-SNAPSHOT.jar
   252518  2016-04-07 00:08   WEB-INF/lib/jackson-core-2.7.3.jar
     2295  2016-04-25 09:21   WEB-INF/lib/spring-boot-starter-logging-1.4.0.BUILD-SNAPSHOT.jar
  1128985  2016-04-25 09:20   WEB-INF/lib/spring-context-4.3.0.BUILD-SNAPSHOT.jar
   855505  2016-04-25 09:21   WEB-INF/lib/spring-boot-autoconfigure-1.4.0.BUILD-SNAPSHOT.jar
     2275  2016-04-25 09:21   WEB-INF/lib/spring-boot-starter-1.4.0.BUILD-SNAPSHOT.jar
     2287  2016-04-25 09:22   WEB-INF/lib/spring-boot-starter-validation-1.4.0.BUILD-SNAPSHOT.jar
    66802  2015-07-10 00:14   WEB-INF/lib/jboss-logging-3.3.0.Final.jar
   470782  2016-04-07 00:07   WEB-INF/lib/logback-core-1.1.7.jar
  1202276  2016-04-07 00:08   WEB-INF/lib/jackson-databind-2.7.3.jar
   704465  2016-02-24 21:13   WEB-INF/lib/hibernate-validator-5.2.4.Final.jar
    23646  2016-04-07 00:14   WEB-INF/lib/log4j-over-slf4j-1.7.21.jar
   905905  2016-04-25 09:20   WEB-INF/lib/spring-webmvc-4.3.0.BUILD-SNAPSHOT.jar
   379313  2016-04-25 09:20   WEB-INF/lib/spring-aop-4.3.0.BUILD-SNAPSHOT.jar
     1186  2016-04-25 09:25   WEB-INF/classes/sample/jetty93/jsp/WelcomeController.class
     1318  2016-04-25 09:25   WEB-INF/classes/sample/jetty93/jsp/SampleJetty93JspApplication.class
      168  2016-04-25 09:25   WEB-INF/classes/application.properties
      409  2016-04-25 09:19   WEB-INF/jsp/welcome.jsp
        0  2016-04-25 09:25   META-INF/maven/
        0  2016-04-25 09:25   META-INF/maven/org.springframework.boot/
        0  2016-04-25 09:25   META-INF/maven/org.springframework.boot/spring-boot-sample-jetty93-jsp/
     2746  2016-04-25 09:19   META-INF/maven/org.springframework.boot/spring-boot-sample-jetty93-jsp/pom.xml
      155  2016-04-25 09:25   META-INF/maven/org.springframework.boot/spring-boot-sample-jetty93-jsp/pom.properties
        0  2016-04-25 09:25   WEB-INF/lib-provided/
     2322  2016-04-25 09:22   WEB-INF/lib-provided/spring-boot-starter-jetty-1.4.0.BUILD-SNAPSHOT.jar
    86386  2016-03-14 08:39   WEB-INF/lib-provided/jetty-servlets-9.3.8.v20160314.jar
    15784  2016-03-14 08:38   WEB-INF/lib-provided/jetty-continuation-9.3.8.v20160314.jar
   138378  2016-03-14 08:31   WEB-INF/lib-provided/jetty-http-9.3.8.v20160314.jar
   106356  2016-03-14 08:32   WEB-INF/lib-provided/jetty-io-9.3.8.v20160314.jar
   111388  2016-03-14 08:34   WEB-INF/lib-provided/jetty-webapp-9.3.8.v20160314.jar
    50313  2016-03-14 08:32   WEB-INF/lib-provided/jetty-xml-9.3.8.v20160314.jar
   116430  2016-03-14 08:34   WEB-INF/lib-provided/jetty-servlet-9.3.8.v20160314.jar
    94508  2016-03-14 08:33   WEB-INF/lib-provided/jetty-security-9.3.8.v20160314.jar
    27855  2016-03-14 08:45   WEB-INF/lib-provided/websocket-server-9.3.8.v20160314.jar
   188483  2016-03-14 08:44   WEB-INF/lib-provided/websocket-common-9.3.8.v20160314.jar
    45302  2016-03-14 08:43   WEB-INF/lib-provided/websocket-api-9.3.8.v20160314.jar
    35971  2016-03-14 08:45   WEB-INF/lib-provided/websocket-client-9.3.8.v20160314.jar
    18180  2016-03-14 08:45   WEB-INF/lib-provided/websocket-servlet-9.3.8.v20160314.jar
    35629  2016-03-14 08:46   WEB-INF/lib-provided/javax-websocket-server-impl-9.3.8.v20160314.jar
    80830  2016-03-14 08:36   WEB-INF/lib-provided/jetty-annotations-9.3.8.v20160314.jar
    55329  2016-03-14 08:36   WEB-INF/lib-provided/jetty-plus-9.3.8.v20160314.jar
    26366  2013-04-26 19:47   WEB-INF/lib-provided/javax.annotation-api-1.2.jar
    53217  2014-03-23 11:14   WEB-INF/lib-provided/asm-5.0.1.jar
    41718  2014-03-23 11:14   WEB-INF/lib-provided/asm-commons-5.0.1.jar
    29038  2014-03-23 11:14   WEB-INF/lib-provided/asm-tree-5.0.1.jar
   157409  2016-03-14 08:46   WEB-INF/lib-provided/javax-websocket-client-impl-9.3.8.v20160314.jar
    36611  2013-05-10 13:09   WEB-INF/lib-provided/javax.websocket-api-1.0.jar
    10647  2016-03-14 08:47   WEB-INF/lib-provided/apache-jsp-9.3.8.v20160314.jar
   417793  2016-03-14 08:31   WEB-INF/lib-provided/jetty-util-9.3.8.v20160314.jar
   485921  2016-03-14 08:33   WEB-INF/lib-provided/jetty-server-9.3.8.v20160314.jar
   189386  2016-01-06 12:35   WEB-INF/lib-provided/jetty-schemas-3.1.jar
    95806  2013-04-25 16:52   WEB-INF/lib-provided/javax.servlet-api-3.1.0.jar
   961568  2015-10-12 13:11   WEB-INF/lib-provided/apache-jsp-8.0.27.jar
   241279  2015-10-12 13:11   WEB-INF/lib-provided/apache-el-8.0.27.jar
  2310271  2015-02-04 17:56   WEB-INF/lib-provided/ecj-4.4.2.jar
        0  2016-04-25 09:25   org/
        0  2016-04-25 09:25   org/springframework/
        0  2016-04-25 09:25   org/springframework/boot/
        0  2016-04-25 09:25   org/springframework/boot/loader/
     1382  2016-04-25 09:20   org/springframework/boot/loader/PropertiesLauncher$1.class
     4599  2016-04-25 09:20   org/springframework/boot/loader/Launcher.class
    16323  2016-04-25 09:20   org/springframework/boot/loader/PropertiesLauncher.class
     1704  2016-04-25 09:20   org/springframework/boot/loader/PropertiesLauncher$FilteredArchive$1.class
     3128  2016-04-25 09:20   org/springframework/boot/loader/ExecutableArchiveLauncher.class
        0  2016-04-25 09:25   org/springframework/boot/loader/util/
     4947  2016-04-25 09:20   org/springframework/boot/loader/util/SystemPropertyUtils.class
     4318  2016-04-25 09:20   org/springframework/boot/loader/LaunchedURLClassLoader.class
     1468  2016-04-25 09:20   org/springframework/boot/loader/MainMethodRunner.class
        0  2016-04-25 09:25   org/springframework/boot/loader/archive/
     1749  2016-04-25 09:20   org/springframework/boot/loader/archive/JarFileArchive$EntryIterator.class
     1438  2016-04-25 09:20   org/springframework/boot/loader/archive/ExplodedArchive$FileEntryIterator$EntryComparator.class
      906  2016-04-25 09:20   org/springframework/boot/loader/archive/Archive.class
      273  2016-04-25 09:20   org/springframework/boot/loader/archive/ExplodedArchive$1.class
      302  2016-04-25 09:20   org/springframework/boot/loader/archive/Archive$Entry.class
     1097  2016-04-25 09:20   org/springframework/boot/loader/archive/JarFileArchive$JarFileEntry.class
     5068  2016-04-25 09:20   org/springframework/boot/loader/archive/ExplodedArchive.class
     1068  2016-04-25 09:20   org/springframework/boot/loader/archive/ExplodedArchive$FileEntry.class
      399  2016-04-25 09:20   org/springframework/boot/loader/archive/Archive$EntryFilter.class
     3792  2016-04-25 09:20   org/springframework/boot/loader/archive/ExplodedArchive$FileEntryIterator.class
     7016  2016-04-25 09:20   org/springframework/boot/loader/archive/JarFileArchive.class
     2052  2016-04-25 09:20   org/springframework/boot/loader/JarLauncher.class
     1669  2016-04-25 09:20   org/springframework/boot/loader/WarLauncher.class
        0  2016-04-25 09:25   org/springframework/boot/loader/data/
     1531  2016-04-25 09:20   org/springframework/boot/loader/data/ByteArrayRandomAccessData.class
     1341  2016-04-25 09:20   org/springframework/boot/loader/data/RandomAccessData$ResourceAccess.class
     3534  2016-04-25 09:20   org/springframework/boot/loader/data/RandomAccessDataFile$DataInputStream.class
      551  2016-04-25 09:20   org/springframework/boot/loader/data/RandomAccessData.class
     2235  2016-04-25 09:20   org/springframework/boot/loader/data/RandomAccessDataFile$FilePool.class
     3390  2016-04-25 09:20   org/springframework/boot/loader/data/RandomAccessDataFile.class
        0  2016-04-25 09:25   org/springframework/boot/loader/jar/
     1427  2016-04-25 09:20   org/springframework/boot/loader/jar/JarFile$3.class
     3212  2016-04-25 09:20   org/springframework/boot/loader/jar/JarURLConnection$JarEntryName.class
    10811  2016-04-25 09:20   org/springframework/boot/loader/jar/JarFileEntries.class
     6555  2016-04-25 09:20   org/springframework/boot/loader/jar/JarURLConnection.class
     4161  2016-04-25 09:20   org/springframework/boot/loader/jar/AsciiBytes.class
     2002  2016-04-25 09:20   org/springframework/boot/loader/jar/JarFile$1.class
     1629  2016-04-25 09:20   org/springframework/boot/loader/jar/ZipInflaterInputStream.class
      430  2016-04-25 09:20   org/springframework/boot/loader/jar/CentralDirectoryVisitor.class
     3362  2016-04-25 09:20   org/springframework/boot/loader/jar/JarEntry.class
     1199  2016-04-25 09:20   org/springframework/boot/loader/jar/JarFile$2.class
     5056  2016-04-25 09:20   org/springframework/boot/loader/jar/CentralDirectoryFileHeader.class
    12422  2016-04-25 09:20   org/springframework/boot/loader/jar/JarFile.class
     1534  2016-04-25 09:20   org/springframework/boot/loader/jar/JarFileEntries$1.class
     2943  2016-04-25 09:20   org/springframework/boot/loader/jar/CentralDirectoryEndRecord.class
      672  2016-04-25 09:20   org/springframework/boot/loader/jar/JarURLConnection$1.class
      306  2016-04-25 09:20   org/springframework/boot/loader/jar/FileHeader.class
     2169  2016-04-25 09:20   org/springframework/boot/loader/jar/Bytes.class
     6365  2016-04-25 09:20   org/springframework/boot/loader/jar/Handler.class
      262  2016-04-25 09:20   org/springframework/boot/loader/jar/JarEntryFilter.class
     4549  2016-04-25 09:20   org/springframework/boot/loader/jar/CentralDirectoryParser.class
     1967  2016-04-25 09:20   org/springframework/boot/loader/jar/JarFileEntries$EntryIterator.class
     1300  2016-04-25 09:20   org/springframework/boot/loader/jar/JarFile$JarFileType.class
     1165  2016-04-25 09:20   org/springframework/boot/loader/ExecutableArchiveLauncher$1.class
     1454  2016-04-25 09:20   org/springframework/boot/loader/PropertiesLauncher$ArchiveEntryFilter.class
     1807  2016-04-25 09:20   org/springframework/boot/loader/PropertiesLauncher$PrefixMatchingArchiveFilter.class
     2335  2016-04-25 09:20   org/springframework/boot/loader/LaunchedURLClassLoader$1.class
     2382  2016-04-25 09:20   org/springframework/boot/loader/PropertiesLauncher$FilteredArchive.class
---------                     -------
 17199994                     140 files

```

**Comment by wilkinsona** [comment]
> Doubly-nested jars are not something that the Java JVM supports natively (rightfully so, as unpacking that is not great

Spring Boot's nested jars don't require any unpacking as the nested entries are stored uncompressed. This allows us to seek to the required location in the outer jar and read the nested jar from there. There are [more details in the documentation](http://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#executable-jar) if you are interested.

**Comment by gregw** [comment]
Fix is committed and will be in next 9.3 release and the upcoming 9.4
